# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentChild.cpp
# Commit: 8ab38c9b4f25
# Full Hash: 8ab38c9b4f25ed4437804b708cb3be993c522400
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-08-26 21:48:35
# Regressor Bug: 1781129
# File Overlap Count: 1
# Description:
#   Bug 1781129 - Part 1: Use BigBuffer for IPCDataTransfer, r=edgar
#   
#   The IPCDataTransfer type is used to transfer Clipboard/Drag & Drop payloads
#   over IPC to allow them to be written to or read from the relevant system
#   interfaces. Previously, the system which was used was somewhat complex, and
# ==============================================================================

diff -r a9c593b5a54a -r 8ab38c9b4f25 dom/ipc/ContentChild.cpp
--- a/dom/ipc/ContentChild.cpp	Fri Aug 26 15:50:18 2022 +0000
+++ b/dom/ipc/ContentChild.cpp	Fri Aug 26 16:08:05 2022 +0000
@@ -3148,7 +3148,7 @@
       for (uint32_t i = 0; i < aTransfers.Length() && !hasFiles; ++i) {
         auto& items = aTransfers[i].items();
         for (uint32_t j = 0; j < items.Length() && !hasFiles; ++j) {
-          if (items[j].data().type() == IPCDataTransferData::TIPCBlob) {
+          if (items[j].data().type() == IPCDataTransferData::TIPCDataTransferBlob) {
             hasFiles = true;
           }
         }
@@ -3171,7 +3171,7 @@
           // We should hide this data from content if we have a file, and we
           // aren't a file.
           bool hidden =
-              hasFiles && item.data().type() != IPCDataTransferData::TIPCBlob;
+              hasFiles && item.data().type() != IPCDataTransferData::TIPCDataTransferBlob;
           dataTransfer->SetDataWithPrincipalFromOtherProcess(
               NS_ConvertUTF8toUTF16(item.flavor()), variant, i,
               nsContentUtils::GetSystemPrincipal(), hidden);